src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java

```diff
@@ -620,14 +631,26 @@ private static boolean isReadable(Path entry) throws IOException {
     }
 
     private SortedMap<String, String> getMutableDependencies() throws IOException {
+        return getMutableDependenciesHashes("", project.getDependencies());
+    }
+
+    private SortedMap<String, String> getMutablePluginDependencies() throws IOException {
+        SortedMap<String, String> fullMap = new TreeMap<>();
+        for (Plugin plugin : project.getBuildPlugins()) {
+            if (config.isPluginDependenciesExcluded(plugin)) {
+                continue;
+            }
+            fullMap.putAll(getMutableDependenciesHashes(
+                    KeyUtils.getVersionlessArtifactKey(repoSystem.createPluginArtifact(plugin)) + "|",
```

Comment:
To keep `compareWithBaseline` feature operational with plugin dependencies, I added an occurrency index to the map key instead of turning the map into a collection.